src/Tests/Microsoft.NET.Build.Tests/GivenWeWantToRequireWindowsForDesktopApps.cs

```diff
@@ -0,0 +1,141 @@
+        [InlineData("UseWindowsForms")]
+        public void It_errors_on_nonwindows_with_the_windows_desktop_sdk(string desktopProperty)
+        {
+            const string ProjectName = "WindowsDesktopSdkErrorTest";
+
+            var asset = CreateWindowsDesktopSdkTestAsset(ProjectName, desktopProperty);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.WindowsDesktopFrameworkRequiresWindows);
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("Microsoft.WindowsDesktop.App")]
+        [InlineData("Microsoft.WindowsDesktop.App.WindowsForms")]
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_builds_on_windows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Pass();
+        }
+
+        [PlatformSpecificTheory(Platform.Linux, Platform.Darwin, Platform.FreeBSD)]
+        [InlineData("Microsoft.WindowsDesktop.App")]
+        [InlineData("Microsoft.WindowsDesktop.App.WindowsForms")]
+        [InlineData("Microsoft.WindowsDesktop.App.WPF")]
+        public void It_errors_on_nonwindows_with_a_framework_reference(string desktopFramework)
+        {
+            const string ProjectName = "WindowsDesktopReferenceErrorTest";
+
+            var asset = CreateWindowsDesktopReferenceTestAsset(ProjectName, desktopFramework);
+            
+            var command = new BuildCommand(Log, Path.Combine(asset.Path, ProjectName));
+
+            command
+                .Execute("/restore")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.WindowsDesktopFrameworkRequiresWindows);
+        }
+
+        private TestAsset CreateWindowsDesktopSdkTestAsset(string projectName, string desktopProperty)
+        {
+            const string tfm = "netcoreapp3.0";
+
+            var testProject = new TestProject()
+            {
+                Name = projectName,
+                TargetFrameworks = tfm,
+                IsSdkProject = true,
+                IsWinExe = true,
+            };
+
+            testProject.AdditionalProperties.Add(desktopProperty, "true");
+
+            return _testAssetsManager.CreateTestProject(testProject)
+                .WithProjectChanges(
+                    (filename, project) =>
+                    {
+                        project.Root.Attribute("Sdk").Value = "Microsoft.NET.Sdk.WindowsDesktop";
```

Comment:
Consider adding a `ProjectSdk` property to TestProject so we have a simpler way to set the SDK.

Reply:
I can also address that; there's one or two other tests I've added that does this, so I can update those at the same time to use the new property.